### PR TITLE
Refactor app result into a class

### PR DESCRIPTION
**Pull request by linsword13**
For now when pass the result into workspace, it still uses the old dict format so that encoders and uploaders work without needing to change. That also ensures existing serialized results do not break.

### CODE REVIEW

File: lib/ramble/ramble/application.py [hunk 1]
```diff
@@ -1866,11 +1851,11 @@ def is_numeric(value):
                 continue
 
             # When strict success is off for repeats (loose success), skip failed exps
-            if exp_inst.results["RAMBLE_STATUS"] == "FAILED":
+            if exp_inst.result.status == "FAILED":
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/application.py]: (This is unrelated to this PR directly) Can we change "FAILED" to `experiment_status.FAILED` ?

File: lib/ramble/ramble/experiment_result.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+
+_DICT_MAPPING = {
+    "name": "name",
```
Comment (douglasjacobsen) [inline lib/ramble/ramble/experiment_result.py:13]: Is it possible to use the namespace attributes where they make sense here?

**Review by douglasjacobsen — COMMENTED** [review]
Two small changes, but otherwise LGTM

File: lib/ramble/ramble/application.py (hunk 1 above)
Comment (linsword13) [reply lib/ramble/ramble/application.py]: Good call, done

File: lib/ramble/ramble/experiment_result.py (hunk 2 above)
Comment (linsword13) [reply lib/ramble/ramble/experiment_result.py:13]: Definitely, that's a great suggestion, done.

**Review by douglasjacobsen — APPROVED** [review]
